File: Pro SPA Development/TheAgencyWithServer/TheAgency/Models/Agent.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace TheAgency.Models
{
    public class Agent
    {       
        [JsonProperty(PropertyName = "agentID")]
        public int AgentID { get; set; }
        [JsonProperty(PropertyName = "codeName")]
        public string CodeName { get; set; }
        [JsonProperty(PropertyName = "firstName")]
        public string FirstName { get; set; }
        [JsonProperty(PropertyName = "lastName")]
        public string LastName { get; set; }
        [JsonProperty(PropertyName = "imagePath")]
        public string ImagePath { get; set; }
        [JsonProperty(PropertyName = "description")]
        public string Description { get; set; }
        [JsonProperty(PropertyName = "tasks")]
        public IList<Task> Tasks { get; set; }
    }
}

File: Pro SPA Development/TheAgencyWithServer/TheAgency/scripts/models/agent.js
define(['backbone', 'collections/tasks'], function (Backbone, Tasks) {
    var agent = Backbone.Model.extend({
        url: function () {
            return '/api/agents/' + this.get('agentID');
        },        
        defaults: {            
            codeName: '',
            firstName: '',
            lastName: '',
            imagePath: '',
            description: '',
            tasks: new Tasks()
        },
        idAttribute: 'agentID',
        validate: function(attrs, options) {
            if (attrs.firstName.length == 0 || attrs.lastName.length == 0) {
                return "Name must include first name and last name!";
            }
        }
    });

    return agent;
});